src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisCodeEditorSessionService.java

```diff
@@ -0,0 +1,161 @@
+                                        AuthorizationCheckService authCheckService, IrisSessionRepository irisSessionRepository,
+                                        GitService gitService, RepositoryService repositoryService) {
+        this.irisConnectorService = irisConnectorService;
+        this.irisMessageService = irisMessageService;
+        this.irisSettingsService = irisSettingsService;
+        this.irisWebsocketService = irisWebsocketService;
+        this.authCheckService = authCheckService;
+        this.irisSessionRepository = irisSessionRepository;
+        this.gitService = gitService;
+        this.repositoryService = repositoryService;
+    }
+
+    /**
+     * Checks if the user has access to the Iris session.
+     * A user has access if they have access to the exercise and the session belongs to them.
+     * If the user is null, the user is fetched from the database.
+     *
+     * @param session The session to check
+     * @param user    The user to check
+     */
+    @Override
+    public void checkHasAccessToIrisSession(IrisSession session, User user) {
+        checkHasAccessToIrisSession(castToSessionType(session, IrisCodeEditorSession.class), user);
+    }
+    
+    private void checkHasAccessToIrisSession(IrisCodeEditorSession session, User user) {
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, session.getExercise(), user);
+        if (!Objects.equals(session.getUser(), user)) {
+            throw new AccessForbiddenException("Iris Code Editor Session", session.getId());
+        }
+    }
+
+    @Override
+    public void checkIsIrisActivated(IrisSession session) {
+        checkIsIrisActivated(castToSessionType(session, IrisCodeEditorSession.class));
+    }
+    
+    private void checkIsIrisActivated(IrisCodeEditorSession ignored) {
+        // Code editor sessions should probably be available for every programming exercise, especially just-created ones
+        // However, we still may want to check something here
+        // FIXME: Await update to Iris settings system
+    }
+    
+    /**
+     * Sends a request containing the current state of the exercise repositories in the code editor
+     * and the entire conversation history to the LLM, and handles the response.
+     *
+     * @param irisSession The session to send the request for
+     */
+    @Override
+    public void requestAndHandleResponse(IrisSession irisSession) {
+        var fromDB = irisSessionRepository.findByIdWithMessagesAndContents(irisSession.getId());
+        if (!(fromDB instanceof IrisCodeEditorSession session)) {
+            throw new BadRequestException("Iris session is not a code editor session");
+        }
+        var exercise = session.getExercise();
+        var params = new HashMap<String, Object>();
+        params.put("chat_history", session.getMessages());
+        params.put("ps", exercise.getProblemStatement());
```

Comment:
Please be more descriptive
```suggestion
        params.put("problem_statement", exercise.getProblemStatement());
```